libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -87,6 +110,37 @@ void StructuralSimplifier::simplify(std::vector<yul::Statement>& _statements)
 					return s;
 				}
 			}
+			else if (isConstExpression(*_switchStmt.expression, constExprVal))
+			{
+				Block matchingCaseBlock;
+				Case* defaultCase = nullptr;
+				bool foundCase = false;
+
+				for (auto& _case: _switchStmt.cases)
+				{
+					if (_case.value && valueOfNumberLiteral(*_case.value) == constExprVal)
+					{
+						matchingCaseBlock = std::move(_case.body);
+						foundCase = true;
+						break;
+					}
+					else if (!_case.value)
+						defaultCase = &_case;
+				}
+
+				if (!foundCase && defaultCase)
```

Comment:
What about using `Block* matchingCaseBlock = nullptr;` and then using this for `foundCase`? It would remove a variable and a move.

Reply:
I thought of that initially but then figured that using `foundCase` was more clear instead of adding double meaning and using pointers.

Changing it back now.